Include/cpython/pyatomic.h

```diff
@@ -0,0 +1,406 @@
+// This header provides cross-platform low-level atomic operations
+// similar to C11 atomics.
+//
+// Operations are sequentially consistent unless they have a suffix indicating
+// otherwise. If in doubt, prefer the sequentially consistent operations.
+//
+// The "_relaxed" suffix for load and store operations indicates the "relaxed"
+// memory order. They don't provide synchronization, but (roughly speaking)
+// guarantee somewhat sane behavior for races instead of undefined behavior.
+// In practice, they correspond to "normal" hardware load and store
+// instructions, so they are almost as inexpensive as plain loads and stores
+// in C.
+//
+// Note that atomic read-modify-write operations like _Py_atomic_add_* return
+// the previous value of the atomic variable, not the new value.
+//
+// See https://en.cppreference.com/w/c/atomic for more information on C11
+// atomics.
+// See https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2055r0.pdf
+// "A Relaxed Guide to memory_order_relaxed" for discussion of and common usage
+// or relaxed atomics.
+
+#ifndef Py_ATOMIC_H
+#define Py_ATOMIC_H
+
+// Atomically adds `value` to `ptr` and returns the previous value
+static inline int
+_Py_atomic_add_int(int *ptr, int value);
+
+static inline int8_t
+_Py_atomic_add_int8(int8_t *ptr, int8_t value);
+
+static inline int16_t
+_Py_atomic_add_int16(int16_t *ptr, int16_t value);
+
+static inline int32_t
+_Py_atomic_add_int32(int32_t *ptr, int32_t value);
+
+static inline int64_t
+_Py_atomic_add_int64(int64_t *ptr, int64_t value);
+
+static inline intptr_t
+_Py_atomic_add_intptr(intptr_t *ptr, intptr_t value);
+
+static inline unsigned int
+_Py_atomic_add_uint(unsigned int *ptr, unsigned int value);
+
+static inline uint8_t
+_Py_atomic_add_uint8(uint8_t *ptr, uint8_t value);
+
+static inline uint16_t
+_Py_atomic_add_uint16(uint16_t *ptr, uint16_t value);
+
+static inline uint32_t
+_Py_atomic_add_uint32(uint32_t *ptr, uint32_t value);
+
+static inline uint64_t
+_Py_atomic_add_uint64(uint64_t *ptr, uint64_t value);
+
+static inline uintptr_t
+_Py_atomic_add_uintptr(uintptr_t *ptr, uintptr_t value);
+
+static inline Py_ssize_t
+_Py_atomic_add_ssize(Py_ssize_t *ptr, Py_ssize_t value);
+
+// Performs an atomic compare-and-exchange. If `*ptr` and `*expected` are
+// equal, then `desired` is stored in `*ptr`. Otherwise `*expected` is updated
+// with the current value of `*ptr`. Returns 1 on success and 0 on failure.
+// These correspond to the "strong" variations of the C11
+// atomic_compare_exchange_* functions.
```

Comment:
There is a single function, no?

```suggestion
// These correspond to the C11
// atomic_compare_exchange_strong() function.
```

atomic_compare_exchange_weak_explicit() has 2 additional arguments (of type memory_order).